### ISSUE TITLE
Enable running multiple suites in parallel

### ISSUE THREAD

**Issue by lahabana** [onsi/ginkgo#999]
In [kuma](https://github.com/kumahq/kuma) we have a large unit test suite which is composed of 1 ginkgo test suite per package.

So far we've been using `go test -p` which runs all these suites in parallel. We're trying to switch to calling ginkgo directly to be able to get test results but this is showing tests suites running [slower](https://github.com/kumahq/kuma/pull/4524#issuecomment-1170903571).

My guess is this: https://onsi.github.io/ginkgo/#running-multiple-suites
> When there are multiple suites to run Ginkgo attempts to compile the suites in parallel but always runs them sequentially. You can control the number of parallel compilation workers using the ginkgo --compilers=N flag, by default Ginkgo runs as many compilers as you have cores.

We have machines with a lot of cores and a lot of tests suites with only a few specs so running the suites sequentially is actually our bottleneck.
What's the reason for not running suites sequentially? 

xref #644 and #971

**Comment by onsi** [comment]
hey there - i'd love to add support for this someday but it's currently unsupported.  two main motivations behind the current behavior were:
- giving users clean non-interleaved and immediate output as the individual specs run.  it's possible to do this in parallel but i haven't built that support yet.
- focusing on running individual suites in parallel was the main motivator - so that users could write big complex integration suites and have them run efficiently.  i didn't want to deal with the complexity of managing parallelizing a given suite *and* parallelizing multiple suites at once.

i have ideas for both of these and a plan to enable concurrent suites someday.  but can't tell you when i'll be able to get to it.  in the meantime `go test` will run suites concurrently and is a workaround though you do lose a few ginkgo-specific niceties when you use `go test` instead of `ginkgo`

**Comment by lahabana** [comment]
Thanks it makes sense and it's indeed a complex feature.

What we've done is: `go test` for devs in their local envs to dev quickly. `ginkgo -p` in CI where we need coverage and test reports.

**Comment by onsi** [comment]
makes sense.  i'd love to be able to get to this someday soon!

**Comment by fiunchinho** [comment]
We have a use case where running parallel suites would be really beneficial. Maybe we are not organizing the tests in the best way, so let me explain.

We want to test the creation of infrastructure in the cloud, which takes time and it's expensive. We want to test different scenarios, meaning several different potential configurations for that infrastructure. We wrote each scenario as a `Orderer` test suite, which creates these cloud resources just one. The tests inside the test suites make assertions on the created resources.
Our problem is that because ginkgo runs suites sequentially, it takes a lot of time to run all the suites. For our use case, there would be no problem in having the test suites running in parallel, because they create their own non-shared infrastructure. Does that make sense?

**Comment by onsi** [comment]
hey @fiunchinho it sounds like your current structure is:

```
/suite-A
  |__ scenario_a_test.go
/suite-B
  |__ scenario_b_test.go
/suite-C
  |__ scenario_c_test.go
...
```

where each `scenario_*_test.go` file looks like:

```go
var _ = Describe("Scenario X", Ordered, func() {
    BeforeAll(func() {
        // do some very expensive cloud infrastructure set up
    })

    It(...)
    It(...)
    It(...)
    It(...)

    AfterAll(func() {
         // clean up
    })
})
```

if so then you can just put all the test files into a single suite and run _that_ with `ginkgo -p`.  Ginkgo will parallellize the different `Ordered` containers with respect to each other.  But will ensure the tests within a given `Ordered` container run in series (in order).

If the repo is open source I'd be happy to take a look.

Now, _if_ each suite is doing something more like:

```go
var _ = BeforeSuite(func() {
    // expensive setup
})

//tests
```

this approach will not work as the `BeforeSuite`s would need to be merged into one (which, perhaps, is viable for you.  Happy to discuss if you can provide more detail.  Also happy to share other potential patterns of this doesn't Just Work™.
